src/SFA.DAS.LoginService.Application/Services/EmailServiceViewModels/InvitationEmailViewModel.cs

```diff
@@ -2,6 +2,7 @@ namespace SFA.DAS.LoginService.Application.Services.EmailServiceViewModels
 {
     public class InvitationEmailViewModel : EmailViewModel
     {
+        public override string EmailAddress { get; set; }
```

Comment:
Why override the property? I don't quite see why we need to do this?

Reply:
Because the Apprentice invitation template requires the `EmailAddress` token.  This is already present in the `EmailViewModel` base, however the reflection code that populates the template `Token` set explicitly ignores anything that isn't a direct member of the most concrete class.

In order to get the token I had to either expose all the base class members to the Token code - this would include `TemplateId` and others.  Or I explicitly included the `EmailAddress` in the concrete class.  In order to do that and not hide the base class property, it has to be `virtual` and `overridden`.

Reply:
ok, I'm beginning to see the problem